### PR TITLE
Remove references to Microsoft.Extensions.Logging

### PR DESCRIPTION
**Pull request by pranavkm**
cc @davidfowl \ @anurse \ @pakrym

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2732)

### CODE REVIEW

File: test/dotnet-projectmodel-server.Tests/DthTestClient.cs [hunk 1]
```diff
@@ -238,7 +234,7 @@ public void Dispose()
                 // Swallow this error for now.
                 // This is a temporary fix for a random failure on CI. The issue happens on Windowx x86
                 // only.
-                _logger.LogError($"Exception thrown durning socket shutting down: {ex.SocketErrorCode}.");
+                Console.Error.WriteLine($"Exception thrown durning socket shutting down: {ex.SocketErrorCode}.");
```
Comment (pakrym) [inline test/dotnet-projectmodel-server.Tests/DthTestClient.cs]: Reporter?

**Comment by pakrym** [comment]
What about actual reference to package?

**Comment by pranavkm** [comment]
There's a cyclic dependency between coreclr.xunit and CLI. We have to compile CoreCLR.xunit against a CLI where it does not reference `ILogger` at which point we can remove the package reference from `Testing.Abstractions`.

File: test/dotnet-projectmodel-server.Tests/DthTestClient.cs (hunk 1 above)
Comment (pranavkm) [inline test/dotnet-projectmodel-server.Tests/DthTestClient.cs]: Just a test.

**Comment by pakrym** [comment]
:shipit:

**Pull request merged**